src/plugins/yy/yyplugin.cpp

```diff
@@ -391,7 +418,12 @@ static void writeId(JsonWriter &json, const char *member, const QString &id, con
 
 static QString spriteId(const Object *object, const QUrl &imageUrl)
 {
-    return optionalProperty(object, "sprite", sanitizeName(QFileInfo(imageUrl.fileName()).completeBaseName()));
+    // If the custom property "sprite" exist use it instead of crawling the file system
+    const QVariant var = object->resolvedProperty("sprite");
+    if (var.isValid())
+        return var.value<QString>();
+    else
+        return sanitizeName(determineResourceId(imageUrl.path()));
```

Comment:
Since `spriteId` is called for each tile object and also for each tile when exporting a tile layer as an asset layer, I think at this point it would be a good idea to cache the result of `determineResourceId(imageUrl.path())` in the `Context` object. It could provide a caching helper like `context.resourceId(const QString &filePath)`, which stores the result in a `QHash<QString, QString> resourceIds`.

Btw, I don't think it's necessary to call `sanitizeName` on the result when that the name is taken from the GameMaker resource. This is only necessary for the fallback case.

Reply:
Done. See comment down below.